mcux/mcux-sdk/drivers/irqsteer/fsl_irqsteer.c

```diff
@@ -144,7 +201,41 @@ IRQn_Type IRQSTEER_GetMasterNextInterrupt(IRQSTEER_Type *base, irqsteer_int_mast
         return (IRQn_Type)irqNum;
     }
 }
+#else  /* **not** __CORTEX_M */
+IRQn_Type IRQSTEER_GetMasterNextInterrupt(IRQSTEER_Type *base, irqsteer_int_master_t intMasterIndex)
+{
+    uint32_t bitOffset, regIndex, chanStatus, sliceNum;
+    int i, j;
+
+    sliceNum = IRQSTEER_GetMasterIrqCount(base, intMasterIndex) / 32 - 1;
+
+    for (i = 0; i <= sliceNum; i++)
+    {
+        bitOffset = 0;
+
+        /* compute the index of the register to be queried */
+        regIndex = FSL_FEATURE_IRQSTEER_CHn_MASK_COUNT - 1 - intMasterIndex * 2 + i;
+
+        /* get register's value */
+        chanStatus = base->CHn_STATUS[regIndex];
+
+        for (j = 0; j < IRQSTEER_INT_SRC_REG_WIDTH; j++)
+        {
+            if ((chanStatus & 1U) != 0U)
+            {
+                return (IRQn_Type)(uint32_t)IRQSTEER_INT_SRC_NUM(regIndex, bitOffset);
+            }
+
+            bitOffset++;
+            chanStatus = chanStatus >> 1U;
+        }
+    }
+
+    return NotAvail_IRQn;
+}
+#endif /* __CORTEX_M */
 
```

Comment:
Is missing IRQSTEER_GetMasterInterruptsStatus() which is here: https://github.com/nxp-mcuxpresso/mcux-sdk/pull/151/files#diff-4e14c6250d0b93ae23138a5afbfff872604276eebc0e3678a388871ee50f6200R238.

Reply:
Weird, I can see it being added just below your comment:

![screen](https://github.com/zephyrproject-rtos/hal_nxp/assets/26043217/2a2f38fe-20e1-4bb0-ae50-3caa44620027)

Am I missing something here?

Reply:
This is on my end:
![image](https://github.com/zephyrproject-rtos/hal_nxp/assets/74421316/9e584a1f-7759-4a9d-8ee8-5a0c721d1cb0)

Reply:
Sorry, my bad... I was on a wrong commit, not all changes.

Reply:
Ah, ok. Was very confused about what was happening here. Would've been a pretty funny issue though